pivy-importer/src/main/groovy/com/linkedin/python/importer/deps/SdistDownloader.groovy

```diff
@@ -26,25 +26,23 @@ class SdistDownloader extends DependencyDownloader {
     static final String SOURCE_DIST_PACKAGE_TYPE = "sdist"
     static final String SOURCE_DIST_ORG = "pypi"
 
-    DependencySubstitution dependencySubstitution
-    boolean latestVersions
-    boolean allowPreReleases
-
     SdistDownloader(String project, File ivyRepoRoot, DependencySubstitution dependencySubstitution,
                     boolean latestVersions, boolean allowPreReleases, boolean lenient) {
 
-        super(project, ivyRepoRoot, lenient)
-        this.dependencySubstitution = dependencySubstitution
-        this.latestVersions = latestVersions
-        this.allowPreReleases = allowPreReleases
+        super(project, ivyRepoRoot, lenient, latestVersions, allowPreReleases, dependencySubstitution)
     }
 
     @Override
     def downloadDependency(String dep) {
         log.info("Pulling in $dep")
         def (String name, String version) = dep.split(":")
 
-        def projectDetails = cache.getDetails(name)
+        def projectDetails = cache.getDetails(name, lenient)
+        // project name is illegal
```

Comment:
The project name here is the module name, if project is illegal, we can't find any information about it on PyPI, we can't find the Json information, thus return details as null. The "illegal" here means is illegal on PyPI since we can't find it on PyPI, which is differentiated from the case we can't find the certain version or certain artifacts whose project name is valid.